Mergesort/MergeSortPJ.cpp

```diff
@@ -0,0 +1,57 @@
+#include<iostream>
+using namespace std;
+
+void merge(int arr[], int front, int rear, int mid){
+  int size1=mid-front+1;  //earlier tried to assign values directly in left and right array but got stuck in assigning values to right array without declaring sizes first
+  int size2=rear-mid;     //also direct assigning worked only for first iteration
+  int L[size1],R[size2];  //declaring left and right arrays
+//assigning values to left array
+  for(int i=0; i<=size1; i++){
+    L[i]=arr[front+i];      //front+1 *front for indicatingthe starting index of original array *+i for incrementing the head fornext iteration
+  }
+//assigning values to right array
+  for(int j=0; j<size2; j++){
+    R[j]=arr[mid+1+j];
+  }
+
+  int i=0,j=0,k=front;
+  while(i<size1 && j<size2){  //run the loop untill both the specifiers are valid
+      if(L[i] < R[j]){
+        arr[k]=L[i];
+        i++;
+        k++;
+      }
+      else{
+      arr[k]=R[j];
+      j++;
+      k++;
+      }
+  }
+}
```

Comment:
merge two arrays at same time sorting!

Reply:
if size1 = 2 and size2 = 3, then while function would terminate earlier but the last elements wouldn't copy in the array.
while(i<size1)
{
   arr[k]=L[i];
    k++;
     i++;
}
and same for size2

Reply:
taking your suggestion i understood the functionality. 
thanks 👍🏽